cmd/tealdbg/debugger.go

```diff
@@ -187,6 +188,28 @@ func (s *session) Step() {
 	s.resume()
 }
 
+func (s *session) StepOver() {
+	currentLine := s.line.Load()
+	// Get the first TEAL opcode in the line
+	currentOp := strings.Fields(s.lines[currentLine])[0]
+
+	func() {
+		s.mu.Lock()
+		defer s.mu.Unlock()
+		// Step over a function call (callsub op).
+		if currentOp == "callsub" {
+			err := s.setBreakpoint(currentLine + 1)
```

Comment:
Is this condition enough to support step over for recursive subroutine calls? I suspect not, since if you only wait for the next invocation of `currentLine + 1`, that will happen before the recursive callsub returns -- you'll be at the right place in the program code, but too deep in the subroutine call stack.

Reply:
Jason is right. The implementation also misses callstack feature in CDT. Maintaining internal callstack would allow 1) display the callstack in CDT 2) correctly process breakpoint only at required call depth.

Reply:
After more thinking and adding the call stack, I think it makes sense to check the call depth for `step over`. I think `resume` should do the same thing by checking 1) previous breakpoints 2) set the earliest possible breakpoint at a point that is closest to the top of the call stack. I'm thinking `step out` should just set a breakpoint where the call frame returns.